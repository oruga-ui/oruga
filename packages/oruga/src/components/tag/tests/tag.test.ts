import { afterEach, describe, expect, test } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OTag from "@/components/tag/Tag.vue";

describe("OTag tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OTag, {
            props: { label: "My Tag" },
        });
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("tag");
        expect(wrapper.text()).toEqual("My Tag");
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("has icon", async () => {
        const wrapper = mount(OTag, {
            props: { label: "My Tag", icon: "home" },
        });

        expect(wrapper.text()).toEqual("My Tag");

        const icon = wrapper.find('[data-oruga="icon"]');
        expect(icon.exists()).toBeTruthy();
    });

    test("render accordingly when has size prop", () => {
        const wrapper = mount(OTag, {
            props: { size: "large" },
        });

        expect(wrapper.classes("o-tag--large")).toBeTruthy();
    });

    test("render accordingly when has variant prop", () => {
        const wrapper = mount(OTag, {
            props: { variant: "danger" },
        });

        expect(wrapper.classes("o-tag--danger")).toBeTruthy();
    });

    test("emit close event", async () => {
        const wrapper = mount(OTag, {
            props: { label: "My Tag", closable: true },
        });

        expect(wrapper.text()).toEqual("My Tag");

        const closeIcon = wrapper.find('[data-oruga="icon"]');
        expect(closeIcon.exists()).toBeTruthy();

        await closeIcon.trigger("click");

        expect(wrapper.emitted("close")).toHaveLength(1);
    });
});
