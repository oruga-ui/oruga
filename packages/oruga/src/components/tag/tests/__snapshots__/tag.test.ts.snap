// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BTag > render correctly 1`] = `
"<div data-oruga="tag" class="o-tag">
  <!--v-if--><span class="o-tag__label"><!--
                @slot Override the tag label
            -->My Tag</span>
  <!--
            @slot Override the close icon
            @binding {close()} close - close function
        -->
  <!--v-if-->
</div>"
`;

exports[`OTag tests > render correctly 1`] = `
"<div data-oruga="tag" class="o-tag">
  <!--v-if--><span class="o-tag__label"><!--
                @slot Override the tag label
            -->My Tag</span>
  <!--
            @slot Override the close icon
            @binding {close()} close - close function
        -->
  <!--v-if-->
</div>"
`;
